packages/langium/src/grammar/workspace/grammar-workspace-manager.ts

```diff
@@ -0,0 +1,57 @@
+/******************************************************************************
+ * Copyright 2022 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import ignore, { Ignore } from 'ignore';
+import { CancellationToken, WorkspaceFolder } from 'vscode-languageserver-protocol';
+import { URI, Utils } from 'vscode-uri';
+import { LangiumSharedServices } from '../../services';
+import { ConfigurationProvider } from '../../workspace/configuration';
+import { FileSystemNode } from '../../workspace/file-system-provider';
+import { DefaultWorkspaceManager } from '../../workspace/workspace-manager';
+import { LangiumGrammarLanguageMetaData } from '../generated/module';
+
+const CONFIG_KEY = 'build';
+
+interface WorkspaceManagerConf {
+    /**
+     * gitignore separated exclusion patterns, separated by comma
+     */
+    ignorePatterns: string
+}
+
+export class LangiumGrammarWorkspaceManager extends DefaultWorkspaceManager {
+
+    protected readonly configurationProvider: ConfigurationProvider;
+    protected matcher: Ignore | null;
```

Comment:
Usually we avoid the usage of `null` and use `undefined` instead.

Reply:
Changed